provider/kubeletconfig/kubeletconfig_resource.go

```diff
@@ -264,7 +350,44 @@ func (k *KubeletConfigResource) Configure(_ context.Context, req resource.Config
 		return
 	}
 
-	collection := connection.ClustersMgmt().V1().Clusters()
-	k.configClient = client.NewKubeletConfigClient(collection)
-	k.clusterWait = common.NewClusterWait(collection)
+	clusterCollection := connection.ClustersMgmt().V1().Clusters()
+	k.clusterClient = common.NewClusterClient(clusterCollection)
+	k.configsClient = client.NewKubeletConfigsClient(clusterCollection)
+	k.clusterWait = common.NewClusterWait(clusterCollection)
+}
+
+func isHCP(ctx context.Context, clusterId string, clusterClient common.ClusterClient) (bool, error) {
+	isHCP := false
+	cluster, err := clusterClient.FetchCluster(ctx, clusterId)
+	if err != nil {
+		return isHCP, err
+	}
+	return cluster.Hypershift().Enabled(), nil
+}
+
+func getKubeletConfigId(ctx context.Context, clusterId string, name string,
+	configsClient client.KubeletConfigsClient) (string, error) {
+	id := ""
+	kubeletConfigs, _, err := configsClient.List(ctx, clusterId, client.NewPaging(1, -1))
+	if err != nil {
+		return id, fmt.Errorf("Cannot list KubeletConfigs for cluster '%s': %v", clusterId, err)
+	}
+	// for classic cluster, there is maxium one kubeletconfig
+	if name == "" {
+		if len(kubeletConfigs) > 0 {
+			return kubeletConfigs[0].ID(), nil
+		} else {
+			return id, fmt.Errorf("Cannot find KubeletConfig for cluster '%s'", clusterId)
+		}
+	}
+	for _, config := range kubeletConfigs {
```

Comment:
Is this loop necessary since there'll only ever be one kubeletConfig anyway?

Reply:
for one HCP cluster, there maybe more than one kubeletConfig. one machinepool can only have single one.